### PR TITLE
Add a recipe for mosaic

### PR DESCRIPTION
**Pull request by xylar**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xylar** [comment]
We are aware of another package on PyPI with the mosaic name, https://pypi.org/project/Mosaic/, but this package seems to have not been updated since 2007.  We currently do not plan to publish on PyPI but would use a name that doesn't conflict on the off chance that we decide to publish there.

File: recipes/mosaic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  summary: Graphics package for MPAS meshes
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  doc_url: https://github.com/E3SM-Project/mosaic/README.md
+  dev_url: https://github.com/E3SM-Project/mosaic/
+
+extra:
+  recipe-maintainers:
+    - andrewdnolan
```
Comment (xylar) [inline recipes/mosaic/meta.yaml:52]: @andrewdnolan, please confirm that you are willing to be a maintainer.

Comment (xylar) [reply recipes/mosaic/meta.yaml:52]: Also, please take a look at the recipe and make sure you don't see obvious mistakes.

File: recipes/mosaic/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - scipy
+    - xarray
+
+test:
+  imports:
+    - mosaic
+  requires:
+    - pip
+  commands:
+    - pip check
```
Comment (xylar) [inline recipes/mosaic/meta.yaml:39]: @andrewdnolan, ideally, we will add more rigorous testing in the future but this should be adequate for now.

**Comment by xylar** [comment]
@conda-forge/help-python, ready for review!

File: recipes/mosaic/meta.yaml (hunk 1 above)
Comment (andrewdnolan) [reply recipes/mosaic/meta.yaml:52]: Yes, I'm happy to be a maintainer.

Everything seems correct to me, at least as far as I know. Thanks for spearheading this!

File: recipes/mosaic/meta.yaml (hunk 2 above)
Comment (andrewdnolan) [reply recipes/mosaic/meta.yaml:39]: I'm working on [some documentation](https://github.com/andrewdnolan/mosaic/tree/docs) now, but tests will be the nest task to tackle!

File: recipes/mosaic/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/mosaic/meta.yaml:52]: Great, hopefully the linter will notice...

**Comment by andrewdnolan** [comment]
Not sure if the linter will pick up on the response to the comment above, so just in case: Yes, I'm happy to be a maintainer.

**Comment by ocefpaf** [comment]
@conda-forge-admin, please lint

**Comment by xylar** [comment]
Thank @ocefpaf and good to know that bot command!

**Pull request merged**